model/src/gateway/event/shard.rs

```diff
@@ -25,7 +25,7 @@ pub struct Connecting {
 #[derive(Clone, Debug, Deserialize, Eq, PartialEq, Serialize)]
 pub struct Disconnected {
     /// The code for the disconnect if not initiated by the host, if any.
-    pub code: Option<u16>,
```

Comment:
Let's keep this as a u16 to avoid both a breaking change and to avoid client failure to receive unknown close codes.